src/pipeline.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, ArgumentTypeError
+from multiprocessing import Process, Pool, Queue
+from socket import gethostname
+
+class PipelineCLIParser(ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.add_argument('--workers', type=positive_int_type, default=1,
+                help="Number of workers to process work", dest='n_workers')
+        self.add_argument('--mode', choices={'crawler', 'ucrawler', 'user'}, default='user',
+                help="Run mode. Either 'crawler', 'ucrawler', or 'user'")
+        self.add_argument('--source', required=True, help='HTTP link to firebase')
+        self.add_argument('--filter', nargs='+', help='A relational condition over metadata that we pull')
+
+def positive_int_type(val):
+    intval = int(val)
+    if intval <= 0:
+        raise ArgumentTypeError("%s is not a positive integer" % val)
+    return intval
```

Comment:
Consider moving positive_int_type() into the PipelineCLIParser class. I don't think it should be used outside of that class.

Reply:
done